packages/python-packages/api-stub-generator/tests/function_parsing_test.py

```diff
@@ -69,3 +79,13 @@ def test_default_values(self):
         assert func_node.args["foo"].default == "1"
         assert func_node.kw_args["bar"].default == "2"
         assert func_node.kw_args["baz"].default == "..."
+
+    def test_typehint_and_docstring_return_types(self):
+        func_node = FunctionNode("test", None, TestClass.with_python2_list_typehint, "test")
+        assert func_node.return_type == "List[TestClass]"
+
+        func_node = FunctionNode("test", None, TestClass.with_python3_list_typehint, "test")
+        assert func_node.return_type == "List[TestClass]"
```

Comment:
Just checking and I'm assuming that this would work when there are multiple inner types? For example: `List[Union[str, int, TestClass]]`

Reply:
Yes it does.